packages/mui-base/src/Select/Root/useSelectRoot.tsx

```diff
@@ -0,0 +1,365 @@
+    controlled: openParam,
+    default: defaultOpen,
+    name: 'Select',
+    state: 'open',
+  });
+
+  const [value, setValueUnwrapped] = useControlled({
+    controlled: valueProp,
+    default: defaultValue,
+    name: 'Select',
+    state: 'value',
+  });
+
+  const setValue: useSelectRoot.ReturnValue['setValue'] = useEventCallback((nextValue, event) => {
+    onValueChange?.(nextValue, event);
+    setValueUnwrapped(nextValue);
+
+    setDirty(nextValue !== validityData.initialValue);
+
+    if (validateOnChange) {
+      fieldControlValidation.commitValidation(nextValue);
+    }
+
+    const index = valuesRef.current.indexOf(nextValue);
+    setSelectedIndex(index);
+    setLabel(labelsRef.current[index] ?? '');
+  });
+
+  useEnhancedEffect(() => {
+    // Wait for the items to have registered their values in `valuesRef`.
+    queueMicrotask(() => {
+      const index = valuesRef.current.indexOf(value);
+      if (index !== -1) {
+        setSelectedIndex(index);
+        setLabel(labelsRef.current[index] ?? '');
+      } else if (value) {
+        warn(
+          `The value \`${typeof value === 'string' ? value : JSON.stringify(value)}\` is not present in the Select options.`,
+        );
+      }
+    });
+  }, [value]);
+
+  const { mounted, setMounted, transitionStatus } = useTransitionStatus(open, animated);
+
+  const runOnceAnimationsFinish = useAnimationsFinished(popupRef);
+
+  const setOpen = useEventCallback((nextOpen: boolean, event?: Event) => {
+    onOpenChange?.(nextOpen, event);
+    setOpenUnwrapped(nextOpen);
+
+    function handleUnmounted() {
+      ReactDOM.flushSync(() => {
+        setMounted(false);
+      });
+
+      // Ensure these aren't flushed synchronously so they occur after the popup has been unmounted.
+      // This ensures the position calculation won't run again (which otherwise leaves
+      // `isPositioned` set to `true` incorrectly once closed).
+      setInnerOffset(0);
+      setInnerFallback(false);
+    }
+
+    if (!nextOpen) {
+      if (animated) {
+        runOnceAnimationsFinish(handleUnmounted);
+      } else {
+        handleUnmounted();
+      }
+    }
+  });
+
+  const floatingRootContext = useFloatingRootContext({
+    elements: {
+      reference: triggerElement,
+      floating: positionerElement,
+    },
+    open,
+    onOpenChange: setOpen,
+  });
+
+  const click = useClick(floatingRootContext, {
+    enabled: !disabled,
+    event: 'mousedown',
+  });
+
+  const dismiss = useDismiss(floatingRootContext, {
+    enabled: !disabled,
+  });
+
+  const role = useRole(floatingRootContext, {
+    role: 'select',
+  });
+
+  const listNavigation = useListNavigation(floatingRootContext, {
+    enabled: !disabled,
+    listRef: elementsRef,
+    activeIndex,
+    selectedIndex,
+    loop,
+    onNavigate: setActiveIndex,
+  });
+
+  const typeahead = useTypeahead(floatingRootContext, {
+    listRef: labelsRef,
+    activeIndex,
+    resetMs: 500,
```

Comment:
The default should be consistent with Menu's typeahead. A shared value referenced from both components will ensure they don't drift apart.